### ISSUE TITLE
Submodule access rights

### ISSUE THREAD

**Issue by ErezBinyamin** [tomasz-lisowski/swsim#2]
When executing step #2 of the Install procedure, the `git clone --recurse-submodules` fails.

The stdout from that command is:
```
Cloning into 'swsim'...
remote: Enumerating objects: 241, done.
remote: Counting objects: 100% (241/241), done.
remote: Compressing objects: 100% (225/225), done.
remote: Total 241 (delta 123), reused 126 (delta 9), pack-reused 0
Receiving objects: 100% (241/241), 100.24 KiB | 3.71 MiB/s, done.
Resolving deltas: 100% (123/123), done.
Submodule 'lib/make-pal' (git@github.com:tomasz-lisowski/make-pal.git) registered for path 'lib/make-pal'
Submodule 'lib/swicc' (git@github.com:tomasz-lisowski/swicc.git) registered for path 'lib/swicc'
Cloning into '/home/dev/Documents/swsim/lib/make-pal'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:tomasz-lisowski/make-pal.git' into submodule path '/home/dev/Documents/swsim/lib/make-pal' failed
Failed to clone 'lib/make-pal'. Retry scheduled
Cloning into '/home/dev/Documents/swsim/lib/swicc'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:tomasz-lisowski/swicc.git' into submodule path '/home/dev/Documents/swsim/lib/swicc' failed
Failed to clone 'lib/swicc'. Retry scheduled
Cloning into '/home/dev/Documents/swsim/lib/make-pal'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:tomasz-lisowski/make-pal.git' into submodule path '/home/dev/Documents/swsim/lib/make-pal' failed
Failed to clone 'lib/make-pal' a second time, aborting
```

**Comment by tomasz-lisowski** [comment]
You are probably used to cloning using HTTPS, but the submodules get cloned using SSH. Take a look at https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-ssh-urls

TL;DR Generate an SSH keypair, add the public key to Github (Settings -> SSH and GPG keys -> New SSH key), then try cloning again.
